src/main.rs

```diff
@@ -13,23 +14,37 @@ use tokio_diesel::AsyncRunQueryDsl;
 use tracing::{error, info};
 use tracing_subscriber::EnvFilter;
 
+use near_chain_configs::GenesisRecords;
+
 use crate::configs::{Opts, SubCommand};
 use crate::db::enums::{AccessKeyAction, ExecutionStatus};
 use crate::db::{establish_connection, AccessKey};
 
 mod configs;
 mod db;
 mod schema;
+mod state_viewer;
 
 const INTERVAL: Duration = Duration::from_millis(100);
 const INDEXER_FOR_WALLET: &str = "indexer_for_wallet";
 
-async fn handle_genesis_public_keys(near_config: near_indexer::NearConfig) {
+
+async fn access_keys_from_state(home_dir: std::path::PathBuf, near_config: near_indexer::NearConfig) {
+    let store = near_store::create_store(&neard::get_store_path(&home_dir));
+
+    let (runtime, state_roots, header) =
+        state_viewer::load_trie_stop_at_height(store, &home_dir, &near_config, state_viewer::LoadTrieMode::Latest);
+
+    let height = header.height();
+    let new_genesis =
+        state_viewer::state_dump(runtime, state_roots.clone(), header, &near_config.genesis.config);
+
+    handle_genesis_public_keys(new_genesis.records, height).await;
+}
+
+async fn handle_genesis_public_keys(records: GenesisRecords, height: u64) {
```

Comment:
```suggestion
async fn handle_genesis_public_keys(records: GenesisRecords, height: near_indexer::near_primitives::types::BlockHeight) {
```

Update to use `BlockHeight` instead of u64 where you use it for height.